client/network/src/protocol.rs

```diff
@@ -1802,6 +1833,14 @@ impl<B: BlockT, H: ExHashT> NetworkBehaviour for Protocol<B, H> {
 		};
 
 		if let CustomMessageOutcome::None = outcome {
+			// Check if any pre-validations are ready to be processed.
+			while let Poll::Ready(Some(result)) = self.block_announce_pre_validation.poll_next_unpin(cx) {
```

Comment:
This is not called when the relay chain block finished import, does it? So I guess there's additional latency introduced. 

Ideally you'd only poll futures that are waiting for a specific block when it enters the blockchain, and not all of them.

Reply:
> Ideally you'd only poll futures that are waiting for a specific block when it enters the blockchain, and not all of them.

Not sure what you mean by this. `FuturesUnordered` will not poll all futures, only the onces that are ready.